Add guidance about SDMs to User Guide (#2565)
pvlib/pvlib-python: docs/sphinx/source/user_guide/modeling_topics/singlediode.rst

* PVsyst SDM: :py:func:`~pvlib.pvsystem.calcparams_pvsyst`
* De Soto SDM: :py:func:`~pvlib.pvsystem.calcparams_desoto`

The second step is to use the output of these functions to compute points on
the SDE's I-V curve. Three points on the SDE I-V curve are typically of special
interest for PV modeling: the maximum power (MP), open circuit (OC), and
short circuit (SC) points. The most convenient function for computing these
points is :py:func:`pvlib.pvsystem.singlediode`. It provides several methods
for solving the SDE:

+------------------+------------+-----------+-------------------------+
| Method           | Type       | Speed     | Guaranteed convergence? |
+==================+============+===========+=========================+
| ``newton``       | iterative  | fast      | no                      |
+------------------+------------+-----------+-------------------------+
| ``brentq``       | iterative  | slow      | yes                     |
+------------------+------------+-----------+-------------------------+
| ``chandrupatla`` | iterative  | fast      | yes                     |
+------------------+------------+-----------+-------------------------+

> | ``lambertw``     | explicit   | medium    | yes                     |

Reviewer: Also, the use of "explicit" here is subject to debate, as I wouldn't be surprised if there is an iterative solver hiding underneath scipy's implementation.

:py:func:`pvlib.pvsystem.singlediode` and then compute a range
of voltages/currents from zero to those extreme points.  This range can then
be used with the above functions to compute the I-V curve.

Special thin film parameters
----------------------------

The PVsyst SDM has two additional, optional parameters to better represent
the behavior of CdTe and a-Si modules.  As these parameters are not included
in the standard SDE, special methods are needed to account for them.
The functions :py:func:`pvlib.pvsystem.max_power_point`,
:py:func:`pvlib.singlediode.bishop88_i_from_v`,
and :py:func:`pvlib.singlediode.bishop88_v_from_i`
can compute the key points and full I-V curves using these parameters.

Reverse bias breakdown
----------------------

Although the standard SDE does not account for reverse bias breakdown, the
following functions can optionally include an extra term for modeling it:
:py:func:`pvlib.pvsystem.max_power_point`,
:py:func:`pvlib.singlediode.bishop88_i_from_v`,
and :py:func:`pvlib.singlediode.bishop88_v_from_i`. 

Model parameter values
----------------------

Despite some models having parameters with similar names, parameter values are

> specific to the model and thus must be produced with the intended model in mind.

Reviewer 2: ```suggestion
specific to each model and thus must be produced with the intended model in mind.
```

> Sometimes sets of parameter values can be read in from external sources, for example:

Reviewer 2: ```suggestion
For some models, sets of parameter values can be read from external sources,
for example:
```

> * CEC SDM parameter database available from :py:func:`~pvlib.pvsystem.retrieve_sam`

Reviewer 2: ```suggestion
* CEC SDM parameter database can be read using :py:func:`~pvlib.pvsystem.retrieve_sam`
```
